layerfetcher/layers.go

```diff
@@ -246,30 +245,30 @@ func (l *Logic) layerHashBlocksReqReceiver(ctx context.Context, msg []byte) []by
 
 	out, err := types.InterfaceToBytes(b)
 	if err != nil {
-		l.log.WithContext(ctx).With().Error("cannot serialize layer blocks response", log.Err(err))
+		l.log.WithContext(ctx).With().Panic("failed to serialize layer blocks response", log.Err(err))
 	}
 
-	return server.SerializeResponse(l.log, out, err)
+	return out, nil
```

Comment:
shouldn't we retiurn some sort of error to the caller here? if serialisation fails `out` will be nil

Reply:
i changed the code so that if we failed to serialize layerBlocks struct, it panics. so at this point `out` is never nil